lib/note-list.jsx

```diff
@@ -1,17 +1,104 @@
 import React, { PropTypes } from 'react';
-import NoteDisplayMixin from './note-display-mixin';
+import { AutoSizer, List } from 'react-virtualized';
 import PublishIcon from './icons/feed';
 import classNames from 'classnames';
-import { get, isEmpty } from 'lodash';
+import { debounce, get, isEmpty } from 'lodash';
 import { connect } from 'react-redux';
 import appState from './flux/app-state';
 import { tracks } from './analytics'
 import filterNotes from './utils/filter-notes';
+import noteTitle from './utils/note-title';
 
-const NoteList = React.createClass( {
+// this constant was determined experimentally
+// and is open to adjustment if it doesn't find
+// the proper balance between visual updates
+// and performance impacts recomputing row heights
+const TYPING_DEBOUNCE_DELAY = 70;
+
+const noteDisplayMaxHeights = {
+	comfy: 24 + 18 + 21,
+	condensed: 24 + 18,
+	expanded: 130,
+};
```

Comment:
Wouldn't it be better if we use constants for those values, like:

```es6
const heights = {
  previewComfy: 21,
  previewExpanded: 88,
  title: 24,
  verticalPadding: 18,
};

const noteDisplayMaxHeights = {
  comfy: heights.title + heights.verticalPadding + heights.previewComfy,
  ...
};
```

Or maybe a simple comment explaining what those numbers represent at a first glance, without the need of testing and inspecting?